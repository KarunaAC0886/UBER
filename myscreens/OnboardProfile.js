import { View, Text, TouchableOpacity, Image,Pressable,ScrollView, StatusBar } from 'react-native'
import React from 'react'
import OnboardBackNavigator from './OnboardBackNavigator';
import styles from './styles';
import { useNavigation } from '@react-navigation/native';

const OnboardProfile = () => {
  const navigation = useNavigation();

  return (
    <ScrollView style={styles.onb_prpl_main_cntr}>

      <View style={styles.onb_prpl_hdr}>
        <OnboardBackNavigator navpath="onboardfirst" />
      </View>
      <View style={styles.onb_prpl_sub_cntr}>
        <View style={styles.onb_prpl_prplimg_cntr}>
          <Image style={styles.onb_prpl_prplimg} source={require('../assets/profile.png')} />
        </View>

        <View >
          <Text style={styles.onb_prpl_prplfrstcntnt}>By tapping the arrow below, you agree to Uberâ€™s Terms of Use and acknowledge that you have read the Privacy Policy.</Text>
        </View>
        <View>
          <Text style={styles.ond_prpl_lstcntnt}>Check the box to indicate that you are atleast 18 years of age, agree to the<TouchableOpacity><Text style={styles.onb_prpl_tndc}>Terms & Conditions</Text></TouchableOpacity>  and acknowledge the <TouchableOpacity><Text style={styles.onb_prpl_pvcyplcy}> Privacy Policy</Text></TouchableOpacity>.</Text>
        </View>

        <Pressable style={styles.onb_prpl_nxt_cntr} onPress={() => navigation.navigate('onboardpayment')}>
          <Text style={styles.onb_prpl_nxtbtn_txt} >Next</Text>
        </Pressable>
      </View>
     <StatusBar/>
    </ScrollView>
  )
}

export default OnboardProfile