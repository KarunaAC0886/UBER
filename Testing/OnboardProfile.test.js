import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import OnboardProfile from '../myscreens/OnboardProfile';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

jest.mock('@expo/vector-icons', () => ({
  AntDesign: 'AntDesign',
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));

describe('Onboard Profile component', () => {
  it('renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<OnboardProfile/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
