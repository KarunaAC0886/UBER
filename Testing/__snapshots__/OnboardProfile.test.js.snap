// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboard Profile component renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "red",
      "height": undefined,
      "width": undefined,
    }
  }
>
  <View>
    <View
      style={
        {
          "height": undefined,
          "width": undefined,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#1A1A1A",
            "height": "100%",
            "justifyContent": "center",
            "paddingLeft": "6%",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <AntDesign
            name="arrowleft"
            style={
              {
                "color": "#ffffff",
                "fontSize": 23,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#1A1A1A",
          "height": undefined,
          "paddingHorizontal": undefined,
          "paddingTop": undefined,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 0,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/profile.png",
            }
          }
          style={
            {
              "height": 130,
              "width": 130,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 20,
              "height": 97,
              "lineHeight": 23.14,
              "marginTop": undefined,
              "width": 341,
            }
          }
        >
          By tapping the arrow below, you agree to Uberâ€™s Terms of Use and acknowledge that you have read the Privacy Policy.
        </Text>
      </View>
      <View>
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 16,
              "lineHeight": 17.58,
              "marginTop": undefined,
            }
          }
        >
          Check the box to indicate that you are atleast 18 years of age, agree to the
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#535AFF",
                  "fontSize": 16,
                  "lineHeight": 17.58,
                  "paddingLeft": 2,
                  "top": undefined,
                }
              }
            >
              Terms & Conditions
            </Text>
          </View>
            and acknowledge the 
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#535AFF",
                  "fontSize": 16,
                  "lineHeight": 17.58,
                  "paddingLeft": 2,
                  "top": undefined,
                }
              }
            >
               Privacy Policy
            </Text>
          </View>
          .
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#000000",
            "borderColor": "#EDF6FF",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 58,
            "justifyContent": "center",
            "marginTop": undefined,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 21,
              "left": 150,
            }
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
